usaspending_api/etl/tests/test_usaspending_assistance_load.py

```diff
@@ -19,6 +22,72 @@ def test_usaspending_assistnace_load():
     # @todo - should there be an assert here?
 
 
+@pytest.mark.django_db(transaction=True)
+def test_award_and_txn_uniqueness():
+    """Each Award should be unique by:
+    - PIID (Contract) or FAIN/URI (Grants)
+    - Parent Award (but grants don't have parents)
+    - Awarding sub-tier Agency
+
+    Each Transaction should be unique by
+    - Award
+    - Modification number"""
+    call_command('loaddata', 'endpoint_fixture_db')
+
+    # agency records from reference_fixture are a prereq
+
+    awards_before_loads = Award.objects.count()
+    txn_before_loads = Transaction.objects.count()
+    call_command('loaddata', 'reference_fixture')
+    filepath = os.path.join(settings.BASE_DIR, 'usaspending_api', 'data',
+                            'usaspending_fin_assist_direct_payments.csv')
+    call_command('load_usaspending_assistance', filepath)
+    base_award_count = Award.objects.count() - awards_before_loads
+    base_txn_count = Transaction.objects.count() - txn_before_loads
+
+    # Re-importing with new FAINS makes new awards
+    def fainchange(row):
+        return _replacer(row, 'federal_award_id', fain_prefixer)
+
+    def new_fain(row):
+        row['federal_award_id'] = '{}-fain2'.format(row['federal_award_id'])
+        return row
+
+    with mutated_csv(filepath, new_fain) as mutant_file:
+        call_command('load_usaspending_assistance', mutant_file.name)
+        print('ok')
+    assert Award.objects.count() == awards_before_loads + (2 * base_award_count
```

Comment:
This is great--thanks for adding such robust tests to this process. Confession: I had to look at the mutator function and this code a few times before grokking what's going on. Do you think it makes sense to augment the comments you already have by adding a note here that says you're now processing the .csv with different FAINSs?